### PR TITLE
leptonic EWK SUSY + susy interface updates

### PR DESCRIPTION
**Pull request by cheidegg**
to be merged BEFORE the JEC PR

### CODE REVIEW

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 ROOT.PyConfig.IgnoreCommandLineOptions = True
 
 MODULES = []
+from CMGTools.TTHAnalysis.tools.multilepFriendTreeProducersToCleanup import MODULES as multiModules
+MODULES += multiModules
```
Comment (gpetruc) [inline TTHAnalysis/macros/prepareEventVariablesFriendTree.py]: Don't include them here, load them with -I

File: RootTools/python/samples/triggers_13TeV_DATA2016.py [hunk 2]
```diff
@@ -152,13 +186,7 @@
...
-triggers_pfht900      = ['HLT_PFHT900_v*']
-triggers_at57         = ['HLT_PFHT200_DiPFJet90_PFAlphaT0p57_v*']
-triggers_at55         = ['HLT_PFHT250_DiPFJet90_PFAlphaT0p55_v*']
-triggers_at53         = ['HLT_PFHT300_DiPFJet90_PFAlphaT0p53_v*']
-triggers_at52         = ['HLT_PFHT350_DiPFJet90_PFAlphaT0p52_v*']
-triggers_at51         = ['HLT_PFHT400_DiPFJet90_PFAlphaT0p51_v*']
-triggers_htmet        = ['HLT_PFHT350_PFMET120_NoiseCleaned_v*']
-triggers_htjet        = ['HLT_PFHT550_4Jet_v*', 'HLT_PFHT650_4Jet_v*', 'HLT_PFHT750_4Jet_v*']
-triggers_mu30ele30    = ['HLT_Mu30_Ele30_CaloIdL_GsfTrkIdVL_v*']
-triggers_doubleele33  = ['HLT_DoubleEle33_CaloIdL_GsfTrkIdVL_v*']
```
Comment (gpetruc) [inline RootTools/python/samples/triggers_13TeV_DATA2016.py:164]: Don't delete triggers from alphaT.

**Review by gpetruc — CHANGES_REQUESTED** [review]
I leave to @peruzzim to review the changes to run_susyMultilepton_cfg.py.
The rest seems to be fine after you fix the other things I have commented on.

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py [hunk 3]
```diff
@@ -44,8 +44,14 @@
 # Lepton Skimming
 ttHLepSkim.minLeptons = 2
 ttHLepSkim.maxLeptons = 999
-#ttHLepSkim.idCut  = ""
-#ttHLepSkim.ptCuts = []
+ttHLepSkim.allowLepTauComb = True
```
Comment (peruzzim) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py]: To be removed (should be false for ttH and SOS, already set to true inside an if later)

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py [hunk 4]
```diff
@@ -44,8 +44,14 @@
...
+
+if analysis=='susy':
+    susyCoreSequence.remove(ttHLepSkim)
+    susyCoreSequence.insert(susyCoreSequence.index(ttHLepSkim)+1,globalSkim)
```
Comment (peruzzim) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py:49]: First ttHLepSkim, then one looks for where ttHLepSkim is. It should be the opposite.

**Review by gpetruc — APPROVED** [review]

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py [hunk 5]
```diff
@@ -337,11 +356,11 @@
 
 from CMGTools.RootTools.samples.triggers_13TeV_DATA2016 import *
 triggerFlagsAna.triggerBits = {
-    'DoubleMu' : triggers_mumu_iso,
+    'DoubleMu' : triggers_mumu_iso ,
```
Comment (peruzzim) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py]: remove space

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py [hunk 6]
```diff
@@ -337,11 +356,11 @@
...
     'DoubleMuSS' : triggers_mumu_ss,
-    'DoubleMuNoIso' : triggers_mumu_noniso,
-    'DoubleEl' : triggers_ee,
-    'MuEG'     : triggers_mue,
+    'DoubleMuNoIso' : triggers_mumu_noniso + triggers_mu27tkmu8,
```
Comment (peruzzim) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py:360]: to be verified why these triggers are not added to the already existing lists

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py [hunk 7]
```diff
@@ -356,7 +375,9 @@
     'SOSDoubleMuLowMET' : triggers_SOS_doublemulowMET,
     'SOSTripleMu' : triggers_SOS_tripleMu,
     'LepTau' : triggers_leptau,
-    'MET' : triggers_metNoMu90_mhtNoMu90,
+    'MET' : triggers_metNoMu90_mhtNoMu90 + triggers_htmet,
+    'htall' : triggers_pfht
```
Comment (peruzzim) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py]: Please use more similar name like simply HT

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py [hunk 8]
```diff
@@ -210,6 +216,20 @@
     susyMultilepton_collections.update({ # for conversion studies
             "selectedPhotons"    : NTupleCollection("PhoGood", photonTypeSusy, 10, help="Selected photons"),
             }) 
+    del susyMultilepton_collections["discardedJets"]
```
Comment (peruzzim) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py:218]: I see that the modification to the cleaning of jets with leptons is in #143. Should be ok when both are merged, to be verified.

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py [hunk 9]
```diff
@@ -453,22 +474,30 @@
...
-#    json = os.environ['CMSSW_BASE']+'/src/CMGTools/TTHAnalysis/data/json/Cert_271036-276811_13TeV_PromptReco_Collisions16_JSON_NoL1T.txt' # 12.9/fb #276811 ICHEP LastRun
-#    json = '/afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/Cert_271036-279931_13TeV_PromptReco_Collisions16_JSON_NoL1T.txt' #24.5/fb
-    json = '/afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/Cert_271036-280385_13TeV_PromptReco_Collisions16_JSON_NoL1T_v2.txt' #27.22/fb
-    processing = "Run2016B-PromptReco-v2"; short = "Run2016B_PromptReco_v2"; run_ranges = [(273150,280385)]; useAAA=False; # -v2 starts from 273150
+    #json = os.environ['CMSSW_BASE']+'/src/CMGTools/TTHAnalysis/data/json/Cert_271036-276811_13TeV_PromptReco_Collisions16_JSON_NoL1T.txt' # 12.9/fb
+    json = os.environ['CMSSW_BASE']+'/src/CMGTools/TTHAnalysis/data/json/Cert_271036-283685_13TeV_PromptReco_Collisions16_JSON_NoL1T.txt' # 33.6/fb
+
+    processing = "Run2016B-23Sep2016-v2"; short = "Run2016B_23Sep2016_v2"; run_ranges = [(272760,273017)]; useAAA=True; # -v2 starts from 272760 to 273017
```
Comment (peruzzim) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py:479]: Not sure we need run ranges here, datasets are already mutually exclusive. Queries with run range are slower and should be avoided if possible.

**Review by peruzzim — CHANGES_REQUESTED** [review]

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py (hunk 7 above)
Comment (cheidegg) [inline TTHAnalysis/cfg/run_susyMultilepton_cfg.py]: This is a name already used in

File: TTHAnalysis/cfg/run_susyMultilepton_cfg.py (hunk 6 above)
Comment (cheidegg) [reply TTHAnalysis/cfg/run_susyMultilepton_cfg.py:360]: see https://github.com/CERN-PH-CMG/cmgtools-lite/blob/d2b37af6e46ff8c74fe6b919fd023f943f8f945b/RootTools/python/samples/triggers_13TeV_DATA2016.py#L159

**Review by peruzzim — APPROVED** [review]

**Comment by gpetruc** [comment]
Hi @cheidegg: do we need to have https://github.com/CERN-PH-CMG/cmg-cmssw/pull/677 merged before this one, or not? 
That PR needs fixes.

File: TTHAnalysis/python/analyzers/globalEventSkimmer.py [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+                    DS=str(m1.group(1))
+                elif m2:
+                    DS=str(m2.group(1))
+                else:
+                    DS=opt[2:] if opt[0]!="!" else opt[3:]
+               
+                if (DS==compName and veto) or (DS!=compName and not veto):
+                    self.vetoDS=True
+
+class globalSkimmer( Analyzer ):
```
Comment (gpetruc) [inline TTHAnalysis/python/analyzers/globalEventSkimmer.py]: it should be called `globalEventSkimmer`, for consistency with the file name and with the name used in `susyCore_modules_cff` (which has `from CMGTools.TTHAnalysis.analyzers.globalEventSkimmer import globalEventSkimmer`)

**Review by gpetruc — CHANGES_REQUESTED** [review]
One last fix is needed

**Review by gpetruc — APPROVED** [review]

**Pull request merged**